File: apps/ios/monocr-ios/HistoryRecord.swift
import Foundation
import SwiftData

@Model
final class HistoryRecord {
    @Attribute(.unique) var id: UUID
    var timestamp: Date
    var fileName: String
    var fileType: String
    var text: String
    var processingTimeMs: Int
    var category: String // "ocr-scan", "contribution", "feedback"
    @Attribute(.externalStorage) var imageData: Data?
    var isSynced: Bool
    var syncAttempts: Int
    var syncError: String?
    
    init(
        id: UUID = UUID(),
        timestamp: Date = Date(),
        fileName: String,
        fileType: String,
        text: String,
        processingTimeMs: Int = 0,
        category: String,
        imageData: Data? = nil,
        isSynced: Bool = false,
        syncAttempts: Int = 0,
        syncError: String? = nil
    ) {
        self.id = id
        self.timestamp = timestamp
        self.fileName = fileName
        self.fileType = fileType
        self.text = text
        self.processingTimeMs = processingTimeMs
        self.category = category
        self.imageData = imageData
        self.isSynced = isSynced
        self.syncAttempts = syncAttempts
        self.syncError = syncError
    }
}

File: apps/android/app/src/main/java/dev/janakhpon/monocr/data/HistoryRecord.kt
package dev.janakhpon.monocr.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(tableName = "history_records")
data class HistoryRecord(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val timestamp: Long = System.currentTimeMillis(),
    val fileName: String,
    val fileType: String,
    val text: String,
    val processingTime: Int,
    val category: String = "ocr-scan",
    val fileUri: String? = null,
    val syncId: String = UUID.randomUUID().toString(),
    val isSynced: Boolean = false,
    val syncAttempts: Int = 0,
    val syncError: String? = null
)
